Automattic/jetpack
# Blocks: Deprecations - ensure supports are exported within larger configuration objects.

Fixes https://github.com/Automattic/jetpack/issues/40055

Note - this may be better dealt with at the Gutenberg end - comment left here: https://github.com/WordPress/gutenberg/pull/63401#discussion_r1832394335

## Proposed changes:

* The reported issue was that with the latest Gutenberg including the affected commit, the Site Editor (and post / page editors) do not load, with console errors displaying. Disabling Jetpack blocks allowed those editors to load again.
* The issue was due to deprecation object exports, and affected four blocks: Payments Intro, Related Posts, Slideshow and Tiled Gallery. 
* The new `stabilizeSupports` function in Gutenberg now expects supports to be part of a larger block configuration object that doesn’t have getter-only behavior.
* Previously, we were using named exports for each property, so each was exported as it's own standalone property (therefore with getter-only behaviour). With the change in this PR, the support export in particular is exported as a block configuration object, compatible with wa `stabilizeSupports` expects.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/40055

## Does this pull request change what data or activity we track or use?


## Testing instructions:

To replicate the issue:

* In trunk locally or using the Jetpack Beta tester (even the latest stable), add the latest Gutenberg nightly plugin which includes the relevant commit. This can be downloaded from here: https://gutenbergtimes.com/need-a-zip-from-master/#nightly
* Ensure Jetpack blocks are enabled: `/wp-admin/admin.php?page=jetpack#/writing`
* Open up the site editor if using a block theme: `/wp-admin/site-editor.php`, or a post / page editor in wp-admin.
* You should notice that the editors do not load, and the console should show the following error: `process-block-type.js:153 Uncaught TypeError: Cannot set property supports of #<Object> which has only a getter`.

To test the fix:

* Follow the same steps as above, after applying this PR.
* The editors should load, with no console errors displayed.
* As this affects deprecation files, it would be wise to test deprecations work as expected. This may be better tested on a JN site:
  * Deactivate the Gutenberg nightly plugin.
  * Download and install an earlier version of Jetpack - 13.3 covers the most recent Payments Intro block deprecation changes, 13.0 covers the most recent Related Post block deprecation changes, 13.4 covers the most recent Slideshow block deprecation changes and 13.5 would cover the most recent Tiled Gallery block deprecation changes (however with TG there are other issues resulting in a broken block when updating version - see below. Best testing for this may just be to check the post and save content in the console matches, minus what is mentioned below).
    * The Payments Intro block is the Payments block without any options chosen.
    * Regarding the Tiled Gallery block, one issue is reported here: p8oabR-1At-p2#comment-8253, the other is related to aria-label, tabindex and role differences. Those are actually covered by the v6 deprecation, but are only showing as differences in the block saves content vs post body because of the other issue - you'll notice that upgrading from 13.5 to the previous latest stable (13.9) works just fine.
    * Previous releases can be selected more easily from the Beta tester, if using the Beta tester plugin
  * With each respective version installed, add a post with the relevant block - Tiled Gallery, Slideshow, Payments Intro or Related Posts.
  * Using the beta tester plugin, apply the branch with this fix.
  * Ensure the post editor is still loading properly.
  * Reactivate the Gutenberg nightly plugin - ensure everything continues to load as before.
